components/properties/src/sets.rs

```diff
@@ -1824,6 +1835,202 @@ pub fn load_for_general_category_group(
     Ok(CodePointSetData::from_code_point_inversion_list(set))
 }
 
+/// Returns a type capable of looking up values for a property specified as a string, as long as it is an
+/// [ECMA262-subset binary property][ecma], using strict matching on the names in the spec.
+///
+/// This handles every property required by ECMA262, except for:
+///
+/// - `Script` and `General_Category`: handle these directly with [`maps::load_general_category()`] and
+///    [`maps::load_script()`].
+///    using property values parsed via [`GeneralCategory::get_name_to_enum_mapper()`] and [`Script::get_name_to_enum_mapper()`]
+///    if necessary.
+/// - `Script_Extensions`: handle this directly using APIs from [`crate::script`], like [`script::load_script_with_extensions_unstable()`]
+/// - `General_Category` mask values: Handle this alongside `General_Category` using [`GeneralCategoryGroup`],
+///    using property values parsed via [`GeneralCategoryGroup::get_name_to_enum_mapper()`] if necessary
+/// - `Assigned`, `All`, and `ASCII` pseudoproperties: Handle these using their equivalent sets:
+///    - `Any` can be expressed as the range `[\u{0}-\u{10FFFF}]`
+///    - `Assigned` can be expressed as the inverse of the set `gc=Cn` (i.e., `\P{gc=Cn}`).
+///    - `ASCII` can be expressed as the range `[\u{0}-\u{7F}]`
+/// - `General_Category` property values can themselves be treated like properties using a shorthand in ECMA262,
+///    simply create the corresponding `GeneralCategory` set.
+///
+/// ```rust
+/// use icu::properties::sets;
+///
+/// let data = sets::load_for_ecma262_unstable(&icu_testdata::unstable(), "Emoji").expect("loading data failed");
```

Comment:
Nit: Emoji is a bad example because people should be using sets of strings for emoji